dev-games/openscenegraph-openmw/openscenegraph-openmw-3.4_p20200425.ebuild

```diff
@@ -0,0 +1,150 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+WX_GTK_VER="3.0-gtk3"
+inherit cmake flag-o-matic wxwidgets
+
+DESCRIPTION="OpenMW-specific fork of OpenSceneGraph"
+HOMEPAGE="https://github.com/OpenMW/osg"
+MY_COMMIT="8b07809fa674ecffe77338aaea2e223b3aadff0e"
+SRC_URI="https://github.com/OpenMW/osg/archive/${MY_COMMIT}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/osg-${MY_COMMIT}"
+
+LICENSE="wxWinLL-3 LGPL-2.1"
+SLOT="0/132" # NOTE: CHECK WHEN BUMPING! Subslot is SOVERSION
+KEYWORDS="~amd64 ~x86"
+IUSE="curl debug doc examples egl ffmpeg fltk fox gdal gif glut gstreamer jpeg
+	lua openexr openinventor osgapps pdf png sdl sdl2 svg tiff
+	truetype vnc wxwidgets xine xrandr zlib"
+
+REQUIRED_USE="
+	sdl2? ( sdl )
+	openexr? ( zlib )
+"
+
+# TODO: COLLADA, FBX, GTA, OpenVRML, Performer, DCMTK
+BDEPEND="
+	app-arch/unzip
+	virtual/pkgconfig
+	doc? ( app-doc/doxygen )
+"
+RDEPEND="
+	!dev-games/openscenegraph
+	media-libs/mesa[egl?]
+	virtual/glu
+	virtual/opengl
+	x11-libs/libSM
+	x11-libs/libXext
+	curl? ( net-misc/curl )
+	examples? (
+		fltk? ( x11-libs/fltk:1[opengl] )
+		fox? ( x11-libs/fox:1.6[opengl] )
+		glut? ( media-libs/freeglut )
+		sdl2? ( media-libs/libsdl2 )
+		wxwidgets? ( x11-libs/wxGTK:${WX_GTK_VER}[opengl,X] )
+	)
+	ffmpeg? ( media-video/ffmpeg:0= )
+	gdal? ( sci-libs/gdal:= )
+	gif? ( media-libs/giflib:= )
+	gstreamer? (
+		media-libs/gstreamer:1.0
+		media-libs/gst-plugins-base:1.0
+	)
+	jpeg? ( virtual/jpeg:0 )
+	lua? ( >=dev-lang/lua-5.1.5:* )
+	openexr? (
+		media-libs/ilmbase:=
+		media-libs/openexr:=
+	)
+	openinventor? ( media-libs/coin )
+	pdf? ( app-text/poppler[cairo] )
+	png? ( media-libs/libpng:0= )
+	sdl? ( media-libs/libsdl )
+	svg? (
+		gnome-base/librsvg
+		x11-libs/cairo
+	)
+	tiff? ( media-libs/tiff:0 )
+	truetype? ( media-libs/freetype:2 )
+	vnc? ( net-libs/libvncserver )
+	xrandr? ( x11-libs/libXrandr )
+	zlib? ( sys-libs/zlib )
+"
+DEPEND="${RDEPEND}
+	x11-base/xorg-proto
```

Comment:
```
--   Found xrandr, version 1.5.2
CMake Warning (dev) at /usr/lib64/cmake/Qt5Core/Qt5CoreMacros.cmake:44 (message):
  qt5_use_modules is not part of the official API, and might be removed in Qt
  6.
Call Stack (most recent call first):
  /usr/lib64/cmake/Qt5Core/Qt5CoreMacros.cmake:431 (_qt5_warn_deprecated)
  src/osgQt/CMakeLists.txt:82 (qt5_use_modules)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find Qt5OpenGL (missing: Qt5OpenGL_DIR)
CMake Error at /usr/lib64/cmake/Qt5Core/Qt5CoreMacros.cmake:451 (message):
  Cannot use "OpenGL" module which has not yet been found.
Call Stack (most recent call first):
  src/osgQt/CMakeLists.txt:82 (qt5_use_modules)


-- Configuring incomplete, errors occurred!
See also "/var/tmp/portage/dev-games/openscenegraph-openmw-3.4_p20200425/work/openscenegraph-openmw-3.4_p20200425_build/CMakeFiles/CMakeOutput.log".
See also "/var/tmp/portage/dev-games/openscenegraph-openmw-3.4_p20200425/work/openscenegraph-openmw-3.4_p20200425_build/CMakeFiles/CMakeError.log".
 * ERROR: dev-games/openscenegraph-openmw-3.4_p20200425::testworld failed (configure phase):
 *   cmake failed
 * 
 * Call stack:
 *     ebuild.sh, line  125:  Called src_configure
 *   environment, line 2052:  Called cmake_src_configure
 *   environment, line  887:  Called die
 * The specific snippet of code:
 *       "${CMAKE_BINARY}" "${cmakeargs[@]}" "${CMAKE_USE_DIR}" || die "cmake failed";
```
Went on after installing `qtopengl`.

Actually could you go through **all** Qt deps required by this package? And then think whether `IUSE="qt5"` should be toggleable in openmw.

` dev-games/openscenegraph-openmw-3.4_p20200425:0/132::testworld  USE="ffmpeg* gif jpeg pdf png sdl svg tiff truetype wxwidgets zlib -curl -debug -doc* -egl -examples* -fltk -fox -gdal -glut -gstreamer -lua -openexr -openinventor -osgapps -sdl2 -vnc -xine -xrandr"`
Even with `-xrandr` it's looking for it...

Reply:
I forgot that osgQt was part of osg in 3.4. Fixed by adding IUSE="qt5". Widgets and OpenGL are the only parts of Qt used: https://github.com/OpenMW/osg/blob/3.4/src/osgQt/CMakeLists.txt#L82

Anyway, OpenMW doesn't need Qt to be enabled here. When using Qt, it uses it itself, for the launcher and for CS. Partially because newer OSG versions don't have osgQt.